docs/resources/dws_ext_data_source.md

```diff
@@ -0,0 +1,120 @@
+---
+subcategory: "Data Warehouse Service (DWS)"
+---
+
+# huaweicloud_dws_ext_data_source
+
+Manages a DWS external data source resource within HuaweiCloud.  
+
+## Example Usage
+
+### Create an external data source to MRS
+
+```hcl
+variable "mrs_id" {}
+variable "mrs_username" {}
+variable "mrs_password" {}
+
+resource "huaweicloud_dws_ext_data_source" "test" {
+  type           = "MRS"
+  name           = "demo"
+  data_source_id = var.mrs_id
+  user_name      = var.mrs_username
+  user_pwd       = var.mrs_password
+  description    = "This is a demo"
+}
+```
+
+### Create an external data source to OBS
+
+```hcl
+variable "dws_agency_obs" {}
+variable "dws_database" {}
+
+resource "huaweicloud_dws_ext_data_source" "test" {
+  type         = "OBS"
+  name         = "demo"
+  user_name    = var.dws_agency_obs
+  connect_info = var.dws_database
+  description  = "This is a demo"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) The DWS cluster ID to which the external data source belongs.
+  The cluster **type** must is **ANALYSIS**.
+
+  Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) The name of the external data source.  
+  The name can contain 1 to 64 characters. Only letters, digits, and underscores (_) are allowed.
+
+  Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) The type of the external data source.  
+  The valid values are **OBS**, and **MRS**.
+
+  -> The OBS data source is supported only in version 8.2.0 or later of cluster.
+
+  Changing this parameter will create a new resource.
```

Comment:
it's better to exchange the order of the two lines.

Reply:
done